scripts/vehicle_node.py

```diff
@@ -0,0 +1,278 @@
+
+        #Publisher for a Float32MultiArray() Message
+        self.pose_array_publisher = rospy.Publisher("/move_vehicle/cmd_array", Float32MultiArray, queue_size=1)
+
+        #Publisher for PointCloud2 terrain map
+        self.point_cloud_publisher = rospy.Publisher("/point_cloud", PointCloud2, latch = True, queue_size=1)
+
+        #Publisher for Vehicle Marker
+        self.grid_publisher = rospy.Publisher('/grid_marker', Marker, queue_size=1)
+
+        #Keeps program running until interrupted
+        rospy.spin()
+
+    def create_gaussian_array(self):
+
+        """
+        Creates an array filled with gaussian noise using the parameters loaded
+        from the parameter server.
+        """
+
+        #Fill array of size l x w with Gaussian Noise.
+        l = int(self.length/self.resolution)
+        w = int(self.width/self.resolution)
+        grid_map = (l,w)
+        gaussian_array = np.random.normal(self.mu, self.sigma, grid_map)
+
+        #Filter the array to smoothen the variation of the noise
+        gaussian_array = gaussian_filter(gaussian_array, self.sigma_filter)
+
+        return gaussian_array
+
+    def velocity_cmd_callback(self, data):
+        
+        """
+        Updates the most recent command velocity to a twist variable
+        """
+
+        self.lock.acquire()
+        try:
+            self.twist = data
+        finally:
+            self.lock.release()
+
+
+    def update_position(self, event):
+
+        """
+        Computes the pose of the vehicle in the terrian map. 
+        """
+
+        #Create a copy of the most recent stored twist data to perform calculations with
+        self.lock.acquire()
+        try:
+            velocity_data = copy.deepcopy(self.twist)
+        finally:
+            self.lock.release()
+
+        #time elapsed since last update position call
+        if hasattr(event, 'last_real'):
+            if event.last_real is None:
+                time = rospy.Duration(0.05)
+            else:
+                time = event.current_real - event.last_real
+        
+        time = time.to_sec()
+
+        #Calculate angle turned in the given time using omega = theta/time
+        angle = velocity_data.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = velocity_data.linear.x*time
+
+        #Calculate yaw of the robot
+        self.vehicle_yaw += angle
+
+        #Calculate vehicle x, y, z position coordinates
+        # TODO recalculate the position based on traveling in a circular arc.
+        self.pose.position.x += (distance)*cos(self.vehicle_yaw)
+        self.pose.position.y += (distance)*sin(self.vehicle_yaw)
+
+        ##########Calculate z position using linear interpolation and create cloud array########
+        
+        #Create range to be used in interpolation function
+        x = np.linspace(0, self.length, int(self.length/self.resolution))
```

Comment:
You shouldn't have to recompute the number of samples here (`int(self.length/self.resolution)`).  You should get this number directly from `gaussian_array`

Reply:
Also, this indexing doesn't look right.  Manually check the case for `length=2` and `resolution=1`.  The difference between two neighboring elements doesn't seem to be equal to `resolution`